lib/listen/event/loop.rb

```diff
@@ -67,7 +67,7 @@ def stop
         return if stopped?
         transition! :stopped
 
-        if @wait_thread.alive?
+        if @wait_thread&.alive?
```

Comment:
Feeback from @ioquatix 

> I think you can just do @wait_thread.join
> the if statement, is a race condition

Building on that, the `return if stopped?` is too. Getting rid of both, in favor of simply `@wait_thread&.join`.